framework/audio/engine/internal/nodes/sanitizernode.cpp

```
@@ -0,0 +1,61 @@
+
+// 24dBFS ceiling, above what volume/pan boost can push a single track to
+static constexpr float HARD_LIMIT = 16.0f;
+
+void SanitizerNode::doSelfProcess(float* buffer, samples_t samplesPerChannel)
+{
+    const size_t sampleCount = static_cast<size_t>(samplesPerChannel) * outputSpec().audioChannelCount;
+
+    for (size_t i = 0; i < sampleCount; ++i) {
+        float x = buffer[i];
+
+        if (!std::isfinite(x)) {
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)